internal/gps/pkgtree/pkgtree.go

```diff
@@ -1046,36 +1043,27 @@ func uniq(a []string) []string {
 // CreateIgnorePrefixTree takes a set of strings to be ignored and returns a
 // trie consisting of strings prefixed with wildcard ignore suffix (*).
 func CreateIgnorePrefixTree(ig map[string]bool) *radix.Tree {
-	var xt *radix.Tree
-
-	// Create a sorted list of all the ignores to have a proper order in
-	// ignores parsing.
-	sortedIgnores := make([]string, len(ig))
+	// Create a sorted list of all the recursive ignores to have a proper
+	// order in ignores parsing.
+	var recursive []string
 	for k := range ig {
-		sortedIgnores = append(sortedIgnores, k)
-	}
-	sort.Strings(sortedIgnores)
-
-	for _, i := range sortedIgnores {
-		// Skip global ignore.
-		if i == "*" {
-			continue
+		// Skip global and non-recursive ignore.
+		if len(k) > 1 && k[len(k)-1] == '*' {
```

Comment:
Filtering on the first loop reduces the slice length for sorting and the second loop, and enables an earlier exit path.